test/document_stores/test_document_store.py

```diff
@@ -119,6 +119,32 @@ def test_init_elastic_doc_store_with_index_recreation():
     assert len(labels) == 0
 
 
+@pytest.mark.elasticsearch
+def test_elasticsearch_eq_filter():
+    documents = [
+        {"content": "some text", "id": "1", "keyword_field": ["x", "y", "z"], "number_field": [1, 2, 3, 4]},
+        {"content": "some text", "id": "2", "keyword_field": ["x", "y", "w"], "number_field": [1, 2, 3]},
+        {"content": "some text", "id": "3", "keyword_field": ["x", "z"], "number_field": [2, 4]},
+        {"content": "some text", "id": "4", "keyword_field": ["x", "y"], "number_field": [2, 3]},
+    ]
+
+    index = "test_elasticsearch_eq_filter"
+    document_store = ElasticsearchDocumentStore(index=index, recreate_index=True)
+    document_store.write_documents(documents)
+
+    filter = {"keyword_field": {"$eq": ["z", "x"]}}
+    filtered_docs = document_store.get_all_documents(index=index, filters=filter)
+    assert len(filtered_docs) == 1
+    assert filtered_docs[0].meta["keyword_field"] == ["x", "z"]
```

Comment:
If the order doesn't matter, I would add another document with `"keyword_field": ["z", "x"]` and then iteratively compare with a set.